Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
> but I find that there's also no particular reason for me to assume that this slight increase in readability to humans varies directly with ease of parse-ability (and readability) to compilers.

The reason I suggested it is not only for readability (if that was the case, I wouldn't have suggested it at all). Why expose a variable to unintentional modification in a scope it is not used in (this applies to any other case, not just this one in particular; yes I know nothing modifies it past this, but that's not the point I'm getting at)? There is no point (IMO) with regards to software robustness/safety aside from the usually heard, "This is how it is in C89 or <insert dated standard from other language here>, therefore I do this habit".

> The same logical issues with direct language parsing having constituted some of the options held back before C99

Right, however, I'm quite sure parsing methods have gotten better since prior to 1999.

> if not also easier for the compiler to parse faster.

This is something I'll need a source on that supports the claim (preferably a modern one).

> unless you can find a reason to suggest that there's anything to lose from my approach

Aside from it being deprecated?